math/src/main/kotlin/ktx/math/ImmutableVector.kt

```diff
@@ -0,0 +1,181 @@
+@file:Suppress("NOTHING_TO_INLINE")
+
+package ktx.math
+
+import com.badlogic.gdx.math.Interpolation
+import com.badlogic.gdx.math.MathUtils
+import com.badlogic.gdx.math.Vector
+import kotlin.math.abs
+import kotlin.math.sqrt
+
+/**
+ * Represent an immutable vector.
+ */
+interface ImmutableVector<T : ImmutableVector<T>> : Comparable<T> {
+
+    /**
+     * Returns the squared euclidean length
+     *
+     * This method is faster than [Vector.len] because it avoids calculating a square root. It is useful for comparisons,
+     * but not for getting exact lengths, as the return value is the square of the actual length.
+     */
+    val len2: Float
+
+    /** Returns the euclidean length */
+    val len: Float get() = sqrt(len2)
+
+    /**
+     * Returns the unit vector of same direction or this vector if it is zero.
+     */
+    val nor: T
+
+    /** Returns whether the length of this vector is smaller than the given [margin] */
+    fun isZero(margin: Float = MathUtils.FLOAT_ROUNDING_ERROR): Boolean
+
+    /** Returns whether this vector is a unit length vector within the given [margin]. (no margin by default) */
+    fun isUnit(margin: Float = MathUtils.FLOAT_ROUNDING_ERROR): Boolean = abs(1f - len2) < margin
+
+    /** Returns the opposite vector of same length */
+    operator fun unaryMinus(): T
+
+    /** Returns the result of subtracting the [other] vector from this vector */
+    operator fun minus(other: T): T
+
+    /** Returns the result of adding the [other] vector to this vector */
+    operator fun plus(other: T): T
+
+    /** Returns the same vector with all members incremented by 1 */
```

Comment:
> Returns the same vector

`Returns a new vector instance`

Please update all docs that mention returning "the same" or "this" vector, as this is misleading.